config/peerpods/podvm/aws-podvm-image-handler.sh

```diff
@@ -343,10 +343,26 @@ function delete_ami_using_id() {
     # AMI_ID shouldn't be empty
     [[ -z "${AMI_ID}" ]] && error_exit "AMI_ID is empty"
 
+    # Retrieve the associated snapshots
+    SNAPSHOT_ID_JSON_PATH='Images[].BlockDeviceMappings[].Ebs.SnapshotId'
+    SNAPSHOT_ID_LIST=$(
+        aws ec2 describe-images \
+        --region "${AWS_REGION}" \
+        --image-ids "${AMI_ID}" \
+        --query "${SNAPSHOT_ID_JSON_PATH}" \
+        --output text
+    )
+
     # Delete the ami
     aws ec2 deregister-image --region "${AWS_REGION}" --image-id "${AMI_ID}" ||
         error_exit "Failed to delete the ami"
 
+    # Delete the associated snapshots
+    for SNAPSHOT_ID in "${SNAPSHOT_ID_LIST}"; do
+        aws ec2 delete-snapshot --region "${AWS_REGION}" --snapshot-id "${SNAPSHOT_ID}" ||
+            error_exit "Failed to delete snapshot '${SNAPSHOT_ID}'"
```

Comment:
This will not expand `${SNAPSHOT_ID}` because of the single quotes

Reply:
> This will not expand `${SNAPSHOT_ID}` because of the single quotes

The whole message sits in a double-quoted string, the single quotes will be preserved and not consumed by the shell in this case. Output will be something like :
```
Failed to delete snapshot 'foo'
```

Reply:
Right! thanks, resolving this conversation